### PR TITLE
Parsed lyrics attribute from a song

### PR DESCRIPTION
**Pull request by pitchumani1989**

### CODE REVIEW

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public String parseSearchTitle(String title, String alternateTitle)
 
     public String parseSearchLyrics(String input)
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: Method parameter is ambiguous. Consider giving a better name like "lyrics".

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 2]
```diff
@@ -42,9 +42,17 @@ public String parseVerseOrder(String input)
...
+    public String parseLyrics(String input)
+    {
+        return parseAttribute(input, "lyrics");
+    }
+
     public String parseAttribute(String input, String attributeName)
     {
-        if (input != "") {
+        if (!input.isEmpty()) {
+            if(attributeName.equalsIgnoreCase("lyrics")) {
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: This logic can be in a separate method.

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 3]
```diff
@@ -64,4 +88,75 @@ public String findMatchingData(String input, String attributeName)
...
+
+            getSongTag(document);
+
+            Transformer transformer = TransformerFactory.newInstance().newTransformer();
+            document.setXmlStandalone(true);
+            transformer.transform(new DOMSource(document), new StreamResult(out));
+
+            System.out.print("Data:"+out.toString());
+        } catch (ParserConfigurationException pce) {
+            pce.printStackTrace();
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: Use logger. (slf4j and logback)

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 4]
```diff
@@ -4,10 +4,29 @@
...
+import org.w3c.dom.Attr;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.CDATASection;
 
 public class SongParser
 {
+    String lyrics, verseOrder;
+
     public String parseTitle(String input)
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: This method can be package-private. Applies to most other methods.

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 5]
```diff
@@ -64,4 +88,75 @@ public String findMatchingData(String input, String attributeName)
...
+            System.out.print("Data:"+out.toString());
+        } catch (ParserConfigurationException pce) {
+            pce.printStackTrace();
+        } catch (TransformerException tfe) {
+            tfe.printStackTrace();
+        }
+        return out.toString();
+    }
+
+    public void getSongTag(Document document)
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: This method should return the "song" element". Change the method name to "getSongElement". Write unit tests.

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 6]
```diff
@@ -64,4 +88,75 @@ public String findMatchingData(String input, String attributeName)
...
+        Element song = document.createElement("song");
+        Attr version = document.createAttribute("version");
+        version.setValue("1.0");
+        song.setAttributeNode(version);
+        document.appendChild(song);
+
+        getLyricsTag(song, document);
+    }
+
+    public void getLyricsTag(Element song, Document document)
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: Do we need to define "song" element as a method parameter? Consider returning "lyrics" element instead.
Change the method name to "getLyricsElement".

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 7]
```diff
@@ -64,4 +88,75 @@ public String findMatchingData(String input, String attributeName)
...
+        return out.toString();
+    }
+
+    public void getSongTag(Document document)
+    {
+        Element song = document.createElement("song");
+        Attr version = document.createAttribute("version");
+        version.setValue("1.0");
+        song.setAttributeNode(version);
+        document.appendChild(song);
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: Do not append here. Instead append this from the calling method.

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 8]
```diff
@@ -64,4 +88,75 @@ public String findMatchingData(String input, String attributeName)
...
+
+        getLyricsTag(song, document);
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: Change this to:

```
song.appendChild(getLyricsTag(document));
```

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 9]
```diff
@@ -64,4 +88,75 @@ public String findMatchingData(String input, String attributeName)
...
+    }
+
+    public void getLyricsTag(Element song, Document document)
+    {
+        Element lyrics = document.createElement("lyrics");
+        song.appendChild(lyrics);
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: Do not append here.

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 10]
```diff
@@ -64,4 +88,75 @@ public String findMatchingData(String input, String attributeName)
...
+        this.lyrics = lyrics;
+        this.verseOrder = verseOrder;
+
+        Writer out = new StringWriter();
+        try {
+            DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+            Document document = docBuilder.newDocument();
+
+            getSongTag(document);
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: Change this to:

```
document.appendChild(getSongTag(document));
```

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 11]
```diff
@@ -64,4 +88,75 @@ public String findMatchingData(String input, String attributeName)
...
+
+    public void getLyricsTag(Element song, Document document)
+    {
+        Element lyrics = document.createElement("lyrics");
+        song.appendChild(lyrics);
+
+        getVerseTag(lyrics, document);
+    }
+
+    public void getVerseTag(Element lyrics, Document document)
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: This method should return "verse" element. Change the method name to "getVerseElement".

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 12]
```diff
@@ -64,4 +85,81 @@ public String findMatchingData(String input, String attributeName)
...
+        try {
+            DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+            Document document = docBuilder.newDocument();
+            document.setXmlStandalone(true);
+
+            Element song = getSongTag(document);
+            document.appendChild(song);
+
+            Element elementLyrics = getLyricsTag(document);
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: "elementLyrics" could be renamed as simply "lyrics".

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 13]
```diff
@@ -64,4 +85,81 @@ public String findMatchingData(String input, String attributeName)
...
+        Attr version = document.createAttribute("version");
+        version.setValue("1.0");
+        song.setAttributeNode(version);
+
+        return song;
+    }
+
+    Element getLyricsTag(Document document)
+    {
+        Element lyrics = document.createElement("lyrics");
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: Make it a one liner.

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 14]
```diff
@@ -4,11 +4,27 @@
...
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Attr;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
 
 public class SongParser
 {
-    public String parseTitle(String input)
+    String lyrics, verseOrder;
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: What is this?

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 15]
```diff
@@ -93,4 +116,136 @@ public void testParseVerseOrder()
...
+    {
+        assertEquals("v", parser.parseVerseType("v1"));
+        assertEquals("c", parser.parseVerseType("c1"));
+    }
+
+    @Test
+    public void testParseVerseLabel()
+    {
+        assertEquals("1", parser.parseVerseType("v1"));
+        assertEquals("2", parser.parseVerseType("v2"));
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java]: Change this test to parse verse type from the text "V2". Please note that "V" is upper case here.

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 16]
```diff
@@ -93,4 +116,136 @@ public void testParseVerseOrder()
...
+        assertEquals("v1", parser.parseVerseOrders("v1 v2 v3 v4")[0]);
+        assertEquals("v2", parser.parseVerseOrders("v1 v2 v3 v4")[1]);
+        assertEquals("v3", parser.parseVerseOrders("v1 v2 v3 v4")[2]);
+        assertEquals("v4", parser.parseVerseOrders("v1 v2 v3 v4")[3]);
+    }
+
+    @Test
+    public void testParseVerseType()
+    {
+        assertEquals("v", parser.parseVerseType("v1"));
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java]: Add a test to parse verse type from "V2".

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 17]
```diff
@@ -93,4 +116,136 @@ public void testParseVerseOrder()
...
+        assertEquals("c", parser.parseVerseType("c1"));
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java]: Add a test to parse verse type from "C2".

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 18]
```diff
@@ -93,4 +116,136 @@ public void testParseVerseOrder()
...
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testParseVerseOrders()
+    {
+        assertEquals("v1", parser.parseVerseOrders("v1 v2 v3 v4")[0]);
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java]: This test should be like this:
```
assertEquals("v1 v2 v3 v4", parser.parseVerseOrders("verseOrder=v1 v2 v3 v4");
```

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 19]
```diff
@@ -93,4 +116,136 @@ public void testParseVerseOrder()
...
+        }
+        assertEquals(expectedLyrics.toString(), parser.getXmlLyrics(lyrics, "v1 o1 c1 o2 o3"));
+    }
+
+    @Test
+    public void testGetVerseTag()
+    {
+        Writer out = new StringWriter();
+        try {
+            DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java]: These three lines are repeating. Consider moving them to "setUp" method.

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 20]
```diff
@@ -93,4 +116,136 @@ public void testParseVerseOrder()
...
+            document.appendChild(verseTag);
+
+            Transformer transformer = TransformerFactory.newInstance().newTransformer();
+            transformer.setOutputProperty("omit-xml-declaration", "yes");
+            transformer.transform(new DOMSource(document), new StreamResult(out));
+            System.out.println(out.toString());
+
+            assertEquals("<verse type=\"\" label=\"\"><![CDATA[data]]></verse>", out.toString());
+        }
+        catch (Exception e) {
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java]: You shouldn't catch exceptions in tests. Consider throwing them.

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 21]
```diff
@@ -93,4 +116,136 @@ public void testParseVerseOrder()
...
+        Writer out = new StringWriter();
+        try {
+            DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+            Document document = docBuilder.newDocument();
+
+            Element verseTag = parser.getVerseTag(document);
+            document.appendChild(verseTag);
+
+            Transformer transformer = TransformerFactory.newInstance().newTransformer();
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java]: This is repeating in many tests. Find a way to avoid duplicate code.

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 22]
```diff
@@ -64,4 +85,81 @@ public String findMatchingData(String input, String attributeName)
...
+            song.appendChild(elementLyrics);
+
+            Element verse = getVerseTag(document);
+            elementLyrics.appendChild(verse);
+
+            Transformer transformer = TransformerFactory.newInstance().newTransformer();
+            transformer.transform(new DOMSource(document), new StreamResult(out));
+
+        } catch (ParserConfigurationException pce) {
+            pce.printStackTrace();
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: Use Logger.

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 23]
```diff
@@ -64,4 +85,81 @@ public String findMatchingData(String input, String attributeName)
...
+            Element elementLyrics = getLyricsTag(document);
+            song.appendChild(elementLyrics);
+
+            Element verse = getVerseTag(document);
+            elementLyrics.appendChild(verse);
+
+            Transformer transformer = TransformerFactory.newInstance().newTransformer();
+            transformer.transform(new DOMSource(document), new StreamResult(out));
+
+        } catch (ParserConfigurationException pce) {
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: "pce" can be simply "e" or "ex".

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 24]
```diff
@@ -64,4 +85,81 @@ public String findMatchingData(String input, String attributeName)
...
+            pce.printStackTrace();
+        } catch (TransformerException tfe) {
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: "tfe" can be simply "e" or "ex".

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 25]
```diff
@@ -64,4 +85,81 @@ public String findMatchingData(String input, String attributeName)
...
+        return out.toString();
+    }
+
+    Element getSongTag(Document document)
+    {
+        Element song = document.createElement("song");
+        Attr version = document.createAttribute("version");
+        version.setValue("1.0");
+        song.setAttributeNode(version);
+
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: Remove empty line.

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 26]
```diff
@@ -53,15 +76,86 @@ public String parseAttribute(String input, String attributeName)
...
+    {
+        String verseOrders[] = parseVerseOrders(verseOrder);
+        Writer out = new StringWriter();
+        try {
+            DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+            Document document = docBuilder.newDocument();
+            document.setXmlStandalone(true);
+
+            Element elementSong = getSongTag(document);
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: "songElement" sounds better.

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 27]
```diff
@@ -93,4 +135,97 @@ public void testParseVerseOrder()
...
+        assertEquals("", parser.parseSearchLyrics(""));
+        assertEquals(searchLyrics, parser.parseSearchLyrics(lyrics));
+    }
+
+    @Test
+    public void testGetXmlLyrics()
+    {
+        String expectedLyrics = "";
+        ClassLoader classLoader = getClass().getClassLoader();
+        try {
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java]: Unit tests typically shouldn't have try-catch blocks. Prefer throwing them.

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 28]
```diff
@@ -93,4 +135,97 @@ public void testParseVerseOrder()
...
+    public void testParseSearchLyrics()
+    {
+        assertEquals("", parser.parseSearchLyrics(""));
+        assertEquals(searchLyrics, parser.parseSearchLyrics(lyrics));
+    }
+
+    @Test
+    public void testGetXmlLyrics()
+    {
+        String expectedLyrics = "";
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java]: Prefer inline declaration.

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 29]
```diff
@@ -93,4 +135,97 @@ public void testParseVerseOrder()
...
+        ClassLoader classLoader = getClass().getClassLoader();
+        try {
+            expectedLyrics = IOUtils.toString(classLoader.getResourceAsStream("sample_lyrics.xml"));
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java]: Consider naming the file after the song, for eg: "lord-i-lift-your-name-on-high.xml". Later you may have to write additional tests for different songs.

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 30]
```diff
@@ -93,4 +135,97 @@ public void testParseVerseOrder()
...
+        }
+        assertEquals(expectedLyrics.toString(), parser.getXmlLyrics(lyrics, "v1 o1 c1 o2 o3"));
+    }
+
+    @Test
+    public void testGetVerseTag() throws TransformerException
+    {
+        Element verseTag = parser.getVerseTag(document);
+        document.appendChild(verseTag);
+        transformer.setOutputProperty("omit-xml-declaration", "yes");
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java]: This is repeating. Why not have this as a separate method in SongParser?

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 31]
```diff
@@ -53,15 +76,86 @@ public String parseAttribute(String input, String attributeName)
...
+        type.setValue("");
+        verse.setAttributeNode(type);
+        Attr label = document.createAttribute("label");
+        label.setValue("");
+        verse.setAttributeNode(label);
+        verse.appendChild(document.createCDATASection("data"));
+        return verse;
+    }
+
+    String[] parseVerseOrders(String verseOrder)
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: Consider renaming the method as "splitVerseOrder".

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 32]
```diff
@@ -93,4 +135,97 @@ public void testParseVerseOrder()
...
+        assertEquals("V1", parser.parseVerseOrders("V1 V2 V3 V4")[0]);
+        assertEquals("V2", parser.parseVerseOrders("V1 V2 V3 V4")[1]);
+        assertEquals("V3", parser.parseVerseOrders("V1 V2 V3 V4")[2]);
+        assertEquals("V4", parser.parseVerseOrders("V1 V2 V3 V4")[3]);
+    }
+
+    @Test
+    public void testParseVerseType()
+    {
+        assertEquals("V", parser.parseVerseType("V1"));
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java]: Please verify from OpenLP db whether the verse type should be lowercase or not. Make appropriate changes accordingly.

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 33]
```diff
@@ -53,15 +76,100 @@ public String parseAttribute(String input, String attributeName)
...
         while (matcher.find()) {
             matchingData = matcher.group(0);
         }
         return matchingData;
     }
-}
+
+    String getXmlLyrics(String lyrics, String verseOrder)
+    {
+        String verseOrders[] = splitVerseOrder(verseOrder);
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: Define these near the place where it is used.

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 34]
```diff
@@ -53,15 +76,100 @@ public String parseAttribute(String input, String attributeName)
...
+            Document document = docBuilder.newDocument();
+            document.setXmlStandalone(true);
+
+            Element songElement = getSongElement(document);
+            document.appendChild(songElement);
+
+            Element lyricsElement = getLyricsElement(document);
+            songElement.appendChild(lyricsElement);
+
+            for (int i = 0; i < verseOrders.length; i++)
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java:114]: Consider having this logic in a separate method.

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 35]
```diff
@@ -93,4 +163,121 @@ public void testParseVerseOrder()
...
+        assertEquals("v", parser.splitVerseType("V1"));
+        assertEquals("v", parser.splitVerseType("V2"));
+        assertEquals("c", parser.splitVerseType("C1"));
+        assertEquals("c", parser.splitVerseType("C2"));
+        assertEquals("o", parser.splitVerseType("O1"));
+        assertEquals("o", parser.splitVerseType("O2"));
+    }
+
+    @Test
+    public void testParseVerseLabel()
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java:251]: Rename the test method.

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 36]
```diff
@@ -93,4 +163,121 @@ public void testParseVerseOrder()
...
+    {
+        assertEquals("1", parser.splitVerseLabel("V1"));
+        assertEquals("2", parser.splitVerseLabel("V2"));
+        assertEquals("1", parser.splitVerseLabel("C1"));
+        assertEquals("2", parser.splitVerseLabel("C2"));
+        assertEquals("1", parser.splitVerseLabel("O1"));
+        assertEquals("2", parser.splitVerseLabel("O2"));
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java:258]: Add a test to assert empty input string.

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 37]
```diff
@@ -93,4 +163,121 @@ public void testParseVerseOrder()
...
+    public void testParseVerseOrders()
+    {
+        assertEquals("V1", parser.splitVerseOrder("V1 V2 V3 V4")[0]);
+        assertEquals("V2", parser.splitVerseOrder("V1 V2 V3 V4")[1]);
+        assertEquals("V3", parser.splitVerseOrder("V1 V2 V3 V4")[2]);
+        assertEquals("V4", parser.splitVerseOrder("V1 V2 V3 V4")[3]);
+    }
+
+    @Test
+    public void testParseVerseType()
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java:239]: Rename the test method.

File: src/test/java/org/worshipsongs/importer/SongParserTest.java [hunk 38]
```diff
@@ -93,4 +163,121 @@ public void testParseVerseOrder()
...
+    {
+        assertEquals("v", parser.splitVerseType("V1"));
+        assertEquals("v", parser.splitVerseType("V2"));
+        assertEquals("c", parser.splitVerseType("C1"));
+        assertEquals("c", parser.splitVerseType("C2"));
+        assertEquals("o", parser.splitVerseType("O1"));
+        assertEquals("o", parser.splitVerseType("O2"));
```
Comment (sskjames) [inline src/test/java/org/worshipsongs/importer/SongParserTest.java:246]: Add a test to assert empty input string.

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 39]
```diff
@@ -53,15 +76,100 @@ public String parseAttribute(String input, String attributeName)
...
+    {
+        return verse.split("")[1].toLowerCase();
+    }
+
+    String splitVerseLabel(String verse)
+    {
+        return verse.split("")[2];
+    }
+
+    String[] splitVerse(String lyrics)
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java:180]: Do we have unit tests for this?

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 40]
```diff
@@ -53,15 +76,100 @@ public String parseAttribute(String input, String attributeName)
...
+    {
+        return lyrics.split("\\[..\\]");
+    }
+
+    public String parseSongBook(String input)
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java]: Unit tests?

File: src/main/java/org/worshipsongs/importer/Song.java [hunk 41]
```diff
@@ -0,0 +1,99 @@
+package org.worshipsongs.importer;
+
+/**
+ * Created by pitchumani on 10/8/15.
+ */
+public class Song
+{
+    String title, alternateTitle, author ,verseOrder ,songBook ,lyrics ,xmlLyrics ,searchTitle , searchLyrics;
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/Song.java:8]: Define each variable in a new line with the access modifier as "private".

File: src/main/java/org/worshipsongs/importer/Song.java [hunk 42]
```diff
@@ -0,0 +1,99 @@
...
+
+    public String getSearchLyrics()
+    {
+        return searchLyrics;
+    }
+
+    public void setSearchLyrics(String searchLyrics)
+    {
+        this.searchLyrics = searchLyrics;
+    }
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/Song.java:98]: Override "equals", "hashCode" and "toString" methods with appropriate unit tests.

File: src/main/java/org/worshipsongs/importer/SongParser.java [hunk 43]
```diff
@@ -53,15 +82,125 @@ public String parseAttribute(String input, String attributeName)
...
+    {
+        return lyrics.split("\\[..\\]");
+    }
+
+    String parseSongBook(String input)
+    {
+        return parseAttribute(input, "songBook");
+    }
+
+    Song parseSong(String input)
```
Comment (sskjames) [inline src/main/java/org/worshipsongs/importer/SongParser.java:190]: This should be a "public" method. Define this as the first method in this class.
